graylog2-web-interface/src/components/systemjobs/SystemJob.jsx

```diff
@@ -3,17 +3,31 @@ import { ProgressBar } from 'react-bootstrap';
 
 import { LinkToNode, Timestamp } from 'components/common';
 
+import ActionsProvider from 'injection/ActionsProvider';
+const SystemJobsActions = ActionsProvider.getActions('SystemJobs');
+
 const SystemJob = React.createClass({
+  _onCancel(job) {
+    return (e) => {
+      e.preventDefault();
+      if (window.confirm(`Are you sure you want to cancel system job "${job.info}"?`)) {
+        SystemJobsActions.cancelJob(job.id);
+      }
+    };
+  },
   render() {
     const job = this.props.job;
     const progress = job.percent_complete < 100 ?
       (<ProgressBar bsStyle="info" active now={job.percent_complete}/>) : <span className="label label-success finished">Finished!</span>;
+    const cancel = job.is_cancelable ? (<span> - <a href="" onClick={this._onCancel(job)}>cancel job</a></span>) : null;
```

Comment:
While not ideal, I think it doesn't look too bad:
<img width="877" alt="Styled button" src="https://cloud.githubusercontent.com/assets/43951/14458433/60b90dbc-00b1-11e6-898d-c376f1d24766.png">

```diff
diff --git a/graylog2-web-interface/src/components/systemjobs/SystemJob.jsx b/graylog2-web-interface/src/components/systemjobs/SystemJob.jsx
index 46345a1..7a7fd92 100644
--- a/graylog2-web-interface/src/components/systemjobs/SystemJob.jsx
+++ b/graylog2-web-interface/src/components/systemjobs/SystemJob.jsx
@@ -19,7 +19,7 @@ const SystemJob = React.createClass({
     const job = this.props.job;
     const progress = job.percent_complete < 100 ?
       (<ProgressBar bsStyle="info" active now={job.percent_complete}/>) : <span className="label label-success finished">Finished!</span>;
-    const cancel = job.is_cancelable ? (<span> - <a href="" onClick={this._onCancel(job)}>cancel job</a></span>) : null;
+    const cancel = job.is_cancelable ? (<button type="button" className="btn btn-primary btn-xs" onClick={this._onCancel(job)}>Cancel Job</button>) : null;

     return (
       <div>
```